pkg/services/quota/context.go

```diff
@@ -0,0 +1,42 @@
+package quota
+
+import (
+	"context"
+	"sync"
+)
+
+type Context struct {
+	context.Context
+	TargetToSrv *TargetToSrv
+}
+
+func FromContext(ctx context.Context, targetToSrv *TargetToSrv) Context {
```

Comment:
The `quota.Context` is used in the store methods [for getting the service for a specific target](https://github.com/grafana/grafana/blob/72fd8063f5d7cb26b80a2c24c2150fabdac3c4c4/pkg/services/quota/quotaimpl/store.go#L93) (in the database we used to store only the target and for backwards compatibility I kept is as is - off course in that case we need to make sure that there are no target conflicts).

I have also considered the option of setting this in the context using `WithContext()` but I decided over the new structure so that I don't have to check everywhere if it's missing. But I don't mind changing it, if the other way around seems more appropriate.

Reply:
Oh since you're passing `quota.Context` instead of `context.Context` it makes sense. I thought you did type cast, i.e. `ctx.(*quota.TargetToSrv)` somewhere. Thanks :+1: